impeller/renderer/backend/vulkan/pipeline_vk.cc

```diff
@@ -20,7 +20,14 @@ PipelineVK::PipelineVK(std::weak_ptr<PipelineLibrary> library,
   is_valid_ = pipeline_ && render_pass_ && layout_ && descriptor_set_layout_;
 }
 
-PipelineVK::~PipelineVK() = default;
+PipelineVK::~PipelineVK() {
+  if (!IsPipelineLive()) {
```

Comment:
Is there a reason to tie this to the lifetime of the `PipelineLibrary`?

AFAICT the constraint on calling these object deletion APIs is specific to Vulkan.  The APIs should be called if the Vulkan device is still valid and must not be called if it was destroyed.

Remove `IsPipelineLive` from the `Pipeline` superclass.  Add a `weak_ptr<DeviceHolder>` to `PipelineVK` and use that to check whether these objects can be deleted.

Reply:
It already existed and mostly represented the same thing.  It could have lead to problems though, done.